src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,98 @@
+package duke.parser;
+
+import duke.data.TaskHandler;
+import duke.data.task.Task;
+import duke.data.exception.DukeException;
+
+import duke.ui.Ui;
+import duke.command.*;
+import duke.storage.Storage;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Class that makes sense of the user commands to Duke.
+ *
+ * @author Won Ye Ji
+ */
+public class Parser {
+    static boolean isTerminated = false;
+    private TaskHandler taskHandler;
+    private Storage storage;
+
+    /**
+     * Initialise variables to run Duke.
+     */
+    public void initialiseDuke() {
+        try {
+            storage = new Storage("./data/taskList.txt");
+            ArrayList<Task> list = storage.loadTasks();
+            taskHandler = new TaskHandler(list);
+        } catch (DukeException e) {
+            System.err.println("Error: Unable to initialise duke");
+            System.out.println(e.getMessage());
+        }
+    }
+
+    /**
+     * Runs Duke.
+     */
+    public void runDuke() {
+        Ui.greet();
+        Scanner sc = new Scanner(System.in);
+        while(!isTerminated) {
```

Comment:
```suggestion
        while (!isTerminated) {
```
I think there should be a whitespace following a java reserved word?